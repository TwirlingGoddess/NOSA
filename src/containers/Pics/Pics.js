import React, { Component } from 'react';
import { connect } from 'react-redux';
import { addDebunks } from '../../actions/index.js';
import { NavLink } from 'react-router-dom';

class Pics extends Component {
  constructor(props) {
    super(props)
    this.state = {
      pic: this.props.images[0],
      debunked: 'nosa'
    }
  }

  handleYessa = (images) => {
    const nextImage = this.props.images.find(image => {
      return image.id > this.state.pic.id
    })
    this.setState({
      pic: nextImage,
      debunked: 'nosa'
    })
  }

  handleNosa = () => {
    this.setState({
      debunked: 'yessa'
    })
    this.props.addDebunks([...this.props.debunks, this.state.pic])
  }

  render () {
    return (
      <div className='Pics'>
        <img className={this.state.debunked} alt='' src={this.state.pic.url}/>
        <aside>
          <button className="yes-button" onClick={this.handleYessa}>{this.state.debunked==='nosa' && 'YESSA' || 'NEXT'}</button>
          <NavLink className="no-button" to="/prints">Print Image</NavLink>
          <button onClick={this.handleNosa}>NOSA</button>
        </aside>
      </div>
    )
  }
}

export const mapStateToProps = (state) => ({
  images: state.images,
  debunks: state.debunks
})

export const mapDispatchToProps = (dispatch) => ({
  addDebunks: ((pic) => dispatch(addDebunks(pic)))
})

export default connect(mapStateToProps, mapDispatchToProps) (Pics)