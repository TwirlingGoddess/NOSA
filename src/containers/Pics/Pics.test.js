import React from 'react';
import { shallow } from 'enzyme';
import Pics, { mapStateToProps, mapDispatchToProps } from './Pics';
import { Provider } from 'react-redux';
import { createStore } from 'redux';
import { rootReducer } from '../../reducers/index.js';
import { addNosa } from '../../actions/index.js';

describe('Pics', () => {
  let wrapper;
  let mockStore = createStore(rootReducer);

  it('should match the snapshot', () => {
    wrapper = (
      <Provider store={mockStore}>
        <Pics />
      </Provider>
    );
    expect(wrapper).toMatchSnapshot()
  })
})