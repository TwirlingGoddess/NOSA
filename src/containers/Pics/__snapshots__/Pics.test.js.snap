// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pics container Pics component should match the snapshot 1`] = `undefined`;
